zebra/zebra_rib.c

```diff
@@ -528,7 +529,7 @@ nexthop_active (afi_t afi, struct route_entry *re, struct nexthop *nexthop, int
       else if (CHECK_FLAG (re->flags, ZEBRA_FLAG_INTERNAL))
 	{
 	  resolved = 0;
-	  for (newhop = match->nexthop; newhop; newhop = newhop->next)
+	  for (ALL_NEXTHOPS_RO(match->nexthop, newhop))
```

Comment:
This is why I don't think we should call the macro ALL_NEXTHOPS_RO ALL_NEXTHOPS_RO anymore since it is not longer READ ONLY